src/core/proto/dst_entry_tcp.cpp

```diff
@@ -313,6 +354,27 @@ ssize_t dst_entry_tcp::fast_send(const iovec *p_iov, const ssize_t sz_iov, xlio_
     return ret;
 }
 
+uint32_t dst_entry_tcp::doca_slow_path(struct pbuf *p, uint16_t flags,
+                                       struct xlio_rate_limit_t &rate_limit)
+{
+    uint32_t ret = 0;
+    bool is_tso_or_zerocopy = !!(flags & (XLIO_TX_PACKET_ZEROCOPY | XLIO_TX_PACKET_TSO));
+
+    m_slow_path_lock.lock();
+    prepare_to_send(rate_limit, true);
+    if (m_b_is_offloaded) {
+        if (is_valid()) {
+            ret = send_doca(p, flags);
+        } else if (!is_tso_or_zerocopy) {
+            // Doesn't look like neigh delay send supports TSO or ZC
```

Comment:
I believe this should not happen with TCP as it's first packet is SYN or SYN-ACK ... Can we print here a warning if it TSO/ZC and neigh is not resolved?